pg_back

```diff
@@ -218,25 +223,29 @@ if [ "${PG_HASPAUSE}" = "1" ]; then
     # pausing the replication replay
     typeset -i PGBK_EXCLLOCKS;
     PGBK_EXCLLOCKS=1;
-    while [ $PGBK_EXCLLOCKS -gt 0  ]
+    PGBK_PAUSE=10
+    while [ $PGBK_EXCLLOCKS -gt 0 ]
     do
-        ${PGBK_BIN}psql -X $OPTS -At -c "SELECT pg_${xlog_or_wal}_replay_pause() where pg_is_in_recovery();" $PGBK_CONNDB
-        if [ $? != 0 ]; then
-            die "could not pause replication replay"
-        fi
         PGBK_EXCLLOCKS=`${PGBK_BIN}psql -X $OPTS -At -c "SELECT count(*) FROM pg_locks WHERE mode = 'AccessExclusiveLock';" $PGBK_CONNDB`
         if [ $? != 0 ]; then
             die "could not get lock information"
         fi
+        
         if [ $PGBK_EXCLLOCKS -gt 0 ]; then
-            ${PGBK_BIN}psql -X $OPTS -At -c "SELECT pg_${xlog_or_wal}_replay_resume();" $PGBK_CONNDB
+            if [ $PGBK_STANDBY_PAUSE_TIMEOUT -le 0 ]; then
+                die "attempt to pause replication exceeded timeout"
+            fi
+
+            echo "The slave database has exclusive locks (vacuum full, truncate or other locking command) running on master"
```

Comment:
It is a good opportunity to rename "master" and "slave" to "primary" and "standby"